warzone/Cards.cpp

```diff
@@ -0,0 +1,442 @@
+
+		for (int i = 0; i < deck->fullDeck->size(); i++)
+		{
+			fullDeck->push_back(deck->fullDeck->at(i));
+		}
+		for (int i = 0; i < deck->drawPile->size(); i++)
+		{
+			drawPile->push_back(deck->drawPile->at(i));
+		}
+	}
+
+	Deck::~Deck()
+	{
+		fullDeck->clear();
+		delete fullDeck;
+		drawPile->clear();
+		delete drawPile;
+	}
+
+	Deck& Deck::operator=(const Deck& deck)
+	{
+		return new Deck(deck);
+	}
+
+	void Deck::add(Card* card)
+	{
+		fullDeck->push_back(card);
+		int* size = (int*)fullDeck->size();
+		drawPile->push_back(size);
+
+	}
+
+	Card* Deck::getFromCatalog(int* cardID)
+	{
+		if ((size_t)cardID < fullDeck->size())
+		{
+			return this->fullDeck->at((size_t)cardID);
+		}
+		return nullptr;
+	}
+
+	int* Deck::draw()
+	{
+		// randomly choose a value from the draw pile
+		srand((unsigned int)time(NULL));
+		int drawIndex = rand() % drawPile->size();
+
+		// Retrieve cardID
+		int* cardID = drawPile->operator[](drawIndex);
+
+		// Get iterator at location for erasing
+		std::vector<int*>::iterator it = drawPile->begin() + drawIndex;
+		drawPile->erase(it);
+
+		return cardID;
+	}
+
+	/*
+	* Takes cardID and adds it to the bottom of the drawpile
+	*/
+	void Deck::returnToDrawPile(int* cardID)
+	{
+		drawPile->push_back(cardID);
+	}
+
+	ostream& operator<<(ostream& out, const Deck& deck)
+	{
+		out << "Here are the contents of the deck" << endl;
+
+		int i = 0;
+		for (std::vector<Card*>::iterator it = deck.fullDeck->begin(); it != deck.fullDeck->end(); ++it) {
+
+			out << *(*it) << "\t\tID: " << i++ << endl;
+		}
+		out << "----end of deck ----" << endl;
+
+		return out;
+	}
+	
+	void Deck::printDrawpile()
+	{
+		cout << "Here are the contents of the drawpile" << endl;
+		cout << "index\tCard ID" << endl;
+
+		int i = 0;
+		for (std::vector<int*>::iterator it = drawPile->begin(); it != drawPile->end(); ++it) {
+
+			int* cardID = (*it);
+			cout << i++ << "\t" << (int)cardID << endl;
+		}
+	}
+
+	// Hand functions
+	Hand::Hand(Deck* deck)
+	{
+		this->deck = deck;
+		this->currentHand = new vector<int*>;
+	}
+
+	Hand::Hand(Hand* hand)
+	{
+		this->deck = new Deck(hand->deck);
+		this->currentHand = new vector<int*>;
+
+		for (int i = 0; i < hand->currentHand->size(); i++)
+		{
+			currentHand->push_back(hand->currentHand->at(i));
+		}
+	}
+
+	Hand::~Hand()
+	{
+		delete deck;
+		currentHand->clear();
+		delete  currentHand;
+	}
+
+	Hand& Hand::operator=(const Hand& hand)
+	{
+		return new Hand(hand);
+	}
+
+	void Hand::addCard(int* cardId)
+	{
+		currentHand->push_back(cardId);
+	}
+
+	/**
+	 * Plays card at hand index
+	 * 
+	 * \param index Index of the card to be played
+	 */
+	void Hand::play(int index) 
+	{
+		if (index < currentHand->size()) {
+
+			std::vector<int*>::iterator it = currentHand->begin() + index;
+			Card* aCard = this->deck->getFromCatalog((*it));
+			aCard->play();
+			deck->returnToDrawPile(*it);
+			currentHand->erase(it);
+		}
+	}
+	vector<int*>* Hand::getCurrentHand()
```

Comment:
Optional of course but you can return a const vector here if you want, i did that a lot for my map. but very optional, it just prevents the vector from being modified